doc/devel/MEP/MEP28.rst

```diff
@@ -0,0 +1,177 @@
+
+.. - **Progress**: Consensus was reached on the mailing list and
+..   implementation work has begun.
+
+.. - **Completed**: The implementation has been merged into master.
+
+.. - **Superseded**: This MEP has been abandoned in favor of another
+     approach.
+
+Branches and Pull requests
+==========================
+
+* `Issue 1109 <https://github.com/matplotlib/matplotlib/issues/1109>`_
+* `Initial thread (mailing list) <https://www.mail-archive.com/matplotlib-devel%40lists.sourceforge.net/msg11325.html>`_
+* `MEP28 PR <https://github.com/matplotlib/matplotlib/pull/4384>`_
+
+
+Abstract
+========
+
+This MEP proposes a new graphical way of specifying simple to moderately
+complex subplot layouts using arrays of strings.
+
+
+Detailed description
+====================
+
+The most popular way of specifying subplots layout is to use the ``subplot``
+function. This ``subplot`` function is rather unintuitive and makes it
+difficult to specify moderately complex subplots. There are alternatives, some
+of them very powerful, but they generally require a lot of code to specifiy
+precisely the layout. There might be room for an in-between method that rely on
+the graphical description of the layout. It can cover all ``subplot`` cases and
+should allow to specify moderately complex subplots.
+
+Examples
+--------
+
+::
+
+  "AB"
+  ┌────────┐┌────────┐
+  │ A      ││ B      │
+  │        ││        │
+  │        ││        │
+  └────────┘└────────┘
+
+  "ABB"
+  ┌──────┐┌──────────┐
+  │ A    ││ B        │
+  │      ││          │
+  │      ││          │
+  └──────┘└──────────┘
+
+  "ABD"
+  "CCD"
+  ┌───────┐┌───────┐┌───────┐
+  │ A     ││ B     ││ D     │
+  │       ││       ││       │
+  │       ││       ││       │
+  └───────┘└───────┘│       │
+  ┌────────────────┐│       │
+  │ C              ││       │
+  │                ││       │
+  │                ││       │
+  └────────────────┘└───────┘
+
+  "AaBb"
+  ┌───────┐┌─┐┌───────┐┌─┐
+  │ A     ││ ││ B     ││ │
+  │       ││ ││       ││ │
+  │       ││ ││       ││ │
+  └───────┘└─┘└───────┘└─┘
+
+  "  b  "
+  "aABCc"
+              ┌───────┐
+              └───────┘
+  ┌─┐┌───────┐┌───────┐┌───────┐┌─┐
+  │ ││ A     ││ B     ││ C     ││ │
+  │ ││       ││       ││       ││ │
+  │ ││       ││       ││       ││ │
+  └─┘└───────┘└───────┘└───────┘└─┘
+
+
+Syntax
+------
+
+A subplot is described using a capital letter, from ``A`` to ``Z`` such that
+only 26 subplots can be specified at once. Colorbars are specified using
+non-capital letters, from ``a`` to ``z``. The main command is the ``layout``
+command that returns a list of axes (subplots and colorbars) following the
+lexicographic order, mixing capital and small letters. For example::
+
+  A,B,C = layout("ABC")
+  A,B,b = layout("ABb")
+
+A layout can span several lines. In such a case, it is necessary the specify the
+layout using an array of strings::
+
+  A,B = layout(["A",
+                "B"])
```

Comment:
Being able to use Python triple-quoted strings here might also be nice:

```
layout("""
AB
CD
""")
```

Here, it would to `s.strip().split('\n')` to get to the "list of strings" as currently specified here.